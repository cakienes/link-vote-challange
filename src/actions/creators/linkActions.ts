import ILink from '../../containers/Interfaces/ILink';
import {
    CREATE_OR_UPDATE_LINK,
    GET_LINKS,
    GET_LINKS_FROM_LOCAL_STORAGE,
    REMOVE_LINK,
} from '../constants/linkConstants';

export const getLinksFromLocalStorage = () => (dispatch: Function) => {
    dispatch({
        type: GET_LINKS_FROM_LOCAL_STORAGE,
    });
};

export const getLinks = () => (dispatch: Function) => {
    dispatch({
        type: GET_LINKS,
    });
};
export const createOrUpdateLink = (link: ILink) => (dispatch: Function) => {
    dispatch({
        type: CREATE_OR_UPDATE_LINK,
        link,
    });
};
export const removeLink = (link: ILink) => (dispatch: Function) => {
    dispatch({
        type: REMOVE_LINK,
        link,
    });
};
