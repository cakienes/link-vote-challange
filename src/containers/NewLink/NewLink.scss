.backLinkHolder {
    font-size: 20px;
    color: #000 !important;
    text-decoration: none !important;
    span {
        margin-right: 10px;
    }
}
