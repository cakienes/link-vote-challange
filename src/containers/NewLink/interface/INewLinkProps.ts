import ILink from '../../Interfaces/ILink';

export default interface INewLinkProps {
    createOrUpdateLink: (link: ILink) => void;
}
