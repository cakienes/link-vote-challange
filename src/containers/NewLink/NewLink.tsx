import React from 'react';
import { connect } from 'react-redux';
import { toastr } from 'react-redux-toastr';
import { Link } from 'react-router-dom';
import uuid from 'uuid';
import { createOrUpdateLink } from '../../actions/creators/linkActions';
import ILink from '../Interfaces/ILink';
import Layout from '../Layout/Layout';
import INewLinkProps from './interface/INewLinkProps';
import './NewLink.scss';
import NewLinkForm from './NewLinkForm/NewLinkForm';

export class NewLink extends React.Component<INewLinkProps, any> {
    render() {
        return (
            <Layout>
                <Link to="/" className="backLinkHolder">
                    <span className="oi oi-arrow-left" />
                    Return to List
                </Link>
                <NewLinkForm onSubmit={this.onFormSubmit} />
            </Layout>
        );
    }

    onFormSubmit = (link: any) => {
        const newLink: ILink = { ...link, createdAt: new Date(), point: 0, id: uuid() };
        this.props.createOrUpdateLink(newLink);

        toastr.success('', link.name + ' Added');
    };
}

const mapStateToProps = (state: any, ownProps: any) => ({
    initialValues: ownProps.data,
});

export const mapDispatchToProps = (dispatch: Function) => {
    return {
        createOrUpdateLink: (link: ILink) => dispatch(createOrUpdateLink(link)),
    };
};
export default connect(
    mapStateToProps,
    mapDispatchToProps,
)(NewLink);
