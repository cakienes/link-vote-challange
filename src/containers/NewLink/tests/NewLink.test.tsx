import { shallow } from 'enzyme';
import React from 'react';
import { createOrUpdateLink } from '../../../actions/creators/linkActions';
import ILink from '../../Interfaces/ILink';
import INewLinkProps from '../interface/INewLinkProps';
import { mapDispatchToProps, NewLink } from '../NewLink';

describe('NewLink.tsx', () => {
    const createOrUpdateLinkMock = jest.fn();
    const newLinkProps: INewLinkProps = {
        createOrUpdateLink: createOrUpdateLinkMock,
    };

    beforeEach(() => {
        jest.spyOn(global.Date, 'now').mockImplementationOnce(() => new Date('2019-06-02T17:13:58.135Z').valueOf());
    });

    it('renders without crashing', async () => {
        expect(shallow(<NewLink {...newLinkProps} />)).toMatchSnapshot();
    });

    describe('mapDispatchToProps', () => {
        it('should dispatch createOrUpdateLink when called', () => {
            const dispatchMock = jest.fn();
            const result = mapDispatchToProps(dispatchMock);

            const dumLink: ILink = {
                createdAt: new Date(),
                id: 'id',
                link: 'link',
                name: 'name',
                point: 0,
                updatedAt: new Date(),
            };
            result.createOrUpdateLink(dumLink);
            expect(JSON.stringify(dispatchMock.mock.calls[0][0])).toEqual(JSON.stringify(createOrUpdateLink(dumLink)));
        });
    });
});
