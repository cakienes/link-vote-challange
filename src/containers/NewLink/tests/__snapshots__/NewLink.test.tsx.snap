// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewLink.tsx renders without crashing 1`] = `ShallowWrapper {}`;
